### PR TITLE
Add utilities functions to time functions

### PR DESCRIPTION
**Pull request by gglanzani**
Reference #193

### CODE REVIEW

File: dask_ml/utils.py [hunk 1]
```diff
@@ -241,6 +248,44 @@ def _format_bytes(n):
     return '%d B' % n
 
 
+@contextlib.contextmanager
+def _timer(name, *args, _logger=None, level='info', **kwargs):
+    """
+    Output execution time of a function to the given logger level
+
+    :param str name: How to name the timer (will be in the logs)
```
Comment (TomAugspurger) [inline dask_ml/utils.py]: We use [numpydoc](http://numpydoc.readthedocs.io/en/latest/format.html) style docstrings. Could you update these to use a `Parameters` section (don't need ad `Returns`).

File: dask_ml/utils.py [hunk 2]
```diff
@@ -241,6 +248,44 @@ def _format_bytes(n):
...
+    :return: None
+    """
+    start = time.perf_counter()
+    _logger =  _logger or logger
+    _logger.info("Starting %s", name)
+    yield
+    stop = time.perf_counter()
+    delta = datetime.timedelta(seconds=stop - start)
+    _logger_level = getattr(_logger, level)
+    _logger_level("Finished %s in %s", name, delta)  # nicer formatting for time.
```
Comment (TomAugspurger) [inline dask_ml/utils.py]: How is this formatted for a timedelta?

Comment (gglanzani) [reply dask_ml/utils.py]:
```
>>> import logging
>>> import datetime
>>> logging.warning("%s", datetime.timedelta(seconds=50))
WARNING:root:0:00:50
>>> logging.warning("%s", datetime.timedelta(seconds=500))
WARNING:root:0:08:20
>>> logging.warning("%s", datetime.timedelta(seconds=5000))
WARNING:root:1:23:20
```

**Comment by gglanzani** [comment]
@TomAugspurger There are usages of `_timed` here and there. I've tried to remove the other usages as well (note that in some cases the time is given back, so there's no way to make the context manager return without some magic — but that would defeat the point).

**Comment by TomAugspurger** [comment]
Pushed some formatting changes.

The usages of `timed` look good. I'll take a closer look tomorrow.

**Comment by gglanzani** [comment]
@TomAugspurger I had to remove `_timed` from the methods.

The issue is that `functools.wraps` doesn't play nicely with methods.

There are three ways out:

- Not use the decorator for the method (chosen here);
- Not use `functools.wraps`, and lose the docstring;
- Use the `wrapt` package, which considerably slows the code down.

Let me know if you're ok with the trade-offs.

File: dask_ml/utils.py [hunk 3]
```diff
@@ -246,6 +253,53 @@ def _format_bytes(n):
...
+    Parameters
+    ----------
+    name : str
+        How to name the timer (will be in the logs)
+    logger : logging.logger
+        The optional logger where to write
+    level : str
+        On which level to log the performance measurement
+    """
+    start = time.perf_counter()
```
Comment (TomAugspurger) [inline dask_ml/utils.py]: Looks like `time.perf_counter` is new in python 3.3, and we support 2.7.

Comment (gglanzani) [reply dask_ml/utils.py]: Yeah, I fixed that. I'm now checking (locally) why the `check_estimator` test is so slow.

Comment (TomAugspurger) [reply dask_ml/utils.py]: I've noticed some strangeness with that test previously. If necessary, we can disable it, but I'm a bit concerned that it's showing this change has a performance impact (though I don't see why that would be).

Comment (gglanzani) [reply dask_ml/utils.py]: @tomaugspurger I'm running it locally with the decorator turned off, but I don't see different results.

I'll test a bit more and report back.

Comment (gglanzani) [reply dask_ml/utils.py]: @TomAugspurger I forgot an indentation in the Lloyd loop :) Fixed now

**Comment by gglanzani** [comment]
@TomAugspurger I've rebased against master.

Local build is green, let's see what py27 holds for me.

File: dask_ml/utils.py (hunk 3 above)
Comment (TomAugspurger) [reply dask_ml/utils.py]: It's entirely possible that was my fault when I did the reformatting. If so then you have my apologies :)

**Comment by TomAugspurger** [comment]
All green, thanks @gglanzani1

**Pull request merged**